src/components/TaskBoard/index.jsx

```diff
@@ -1,31 +1,59 @@
-import React, { useState } from "react";
-import { useSelector } from "react-redux";
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
 import AddTaskCard from "components/TaskCard/AddTaskCard.jsx";
 import style from "components/TaskBoard/index.module.scss";
+import Pagination from "components/Pagination";
+import {
+  PAGINATION_LIMIT,
+  TEXT_SHOW_MORE,
+  TEXT_SHOW_LESS,
+} from "utils/constant";
+import { paginationLimitUpdate } from "store/actions/";
 import TaskList from "components/TaskList";
 
 const TaskBoard = () => {
-  const [showCreateCard, setShowCreateCard] = useState(false);
+  const [isCardCreated, setIsCardCreated] = useState(false);
+  const paginationLength = useSelector((state) => state.paginationLength);
   const tasks = useSelector((state) => state.todo);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (isCardCreated) {
+      dispatch(paginationLimitUpdate(paginationLength - 1));
+    } else {
+      if (paginationLength != PAGINATION_LIMIT) {
+        dispatch(paginationLimitUpdate(paginationLength + 1));
+      }
+    }
+  }, [isCardCreated]);
 
   const onCreateButtonClick = () => {
-    setShowCreateCard(!showCreateCard);
+    setIsCardCreated(!isCardCreated);
   };
 
+  const isPaginationButtonVisible =
+    tasks.length + isCardCreated > PAGINATION_LIMIT;
+
   return (
     <div className={style.container}>
       <h1>Add Task</h1>
       <div>
-        <button disabled={showCreateCard} onClick={onCreateButtonClick}>
+        <button disabled={isCardCreated} onClick={onCreateButtonClick}>
```

Comment:
```suggestion
        <button disabled={isCardCreated} onClick={handleCreateClick}>
```
[reference](https://react.dev/learn/responding-to-events)

![Screenshot 2023-05-01 at 11 20 07 AM](https://user-images.githubusercontent.com/128278816/235410541-3fd207a0-6c22-46da-8d6c-22d7ed9c5672.png)